spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/config/LoadBalancerAutoConfiguration.java

```diff
@@ -49,6 +51,15 @@ public LoadBalancerAutoConfiguration(
 		this.configurations = configurations;
 	}
 
+	// Adding a @Primary bean to avoid NoUniqueBeanDefinition exception during constructor
```

Comment:
Hmm. This shouldn't be needed. `LoadBalancerAutoConfiguration` should only be once.

Reply:
There was a problem during the automated constructor argument resolution by framework when `@Primary` bean was not defined here. Was getting:

`"Error creating bean with name 'healthCheckServiceInstanceListSupplier' defined in io.github.olgamaciaszek.excluded.CustomLoadBalancerConfiguration: Unsatisfied dependency expressed through method 'healthCheckServiceInstanceListSupplier' parameter 2; nested exception is org.springframework.beans.factory.NoUniqueBeanDefinitionException: No qualifying bean of type 'org.springframework.cloud.client.loadbalancer.reactive.LoadBalancerProperties' available: expected single matching bean but found 2: spring.cloud.loadbalancer-org.springframework.cloud.client.loadbalancer.reactive.LoadBalancerProperties,loadBalancerProperties"` as part of the `ConstructorResolver` exectution.

I was not entirely sure what was causing that as wherever that bean is explicitly instantiated, it is always annotated with `@ConditionalOnMissingBean`.

Any thoughts @spencergibb ?

Reply:
I see. It's already defined in `LoadBalancerClientConfiguration`. Can you eliminate this bean definition?

Reply:
Yes, that's right. Thought `@ConditionalOnMissingBean` would handle any such thing, but it seems it gets instantiated before and then the one generated from `@ConfigurationProperties` clashes.